Index: leocad-18.02/qt/leocad.desktop
===================================================================
--- leocad-18.02.orig/qt/leocad.desktop
+++ leocad-18.02/qt/leocad.desktop
@@ -9,7 +9,7 @@ Comment[pt_BR]=Criar modelos virtuais de
 Exec=leocad %f
 Terminal=false
 Type=Application
-Icon=leocad.svg
+Icon=leocad
 MimeType=application/vnd.leocad;application/x-ldraw;application/x-multi-part-ldraw;application/x-ldlite;
 Categories=Graphics;3DGraphics;Education;
 Keywords=CAD;LEGO;LDraw;
