--- ParaView-v5.5.2.orig/Qt/Components/pqChangeInputDialog.cxx
+++ ParaView-v5.5.2/Qt/Components/pqChangeInputDialog.cxx
@@ -45,6 +45,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include "vtkSMProxy.h"
 #include "vtkSmartPointer.h"
 
+#include <QHeaderView>
 #include <QItemSelectionModel>
 #include <QLineEdit>
 #include <QRadioButton>
--- ParaView-v5.5.2.orig/Qt/Components/pqCinemaTrackSelection.cxx
+++ ParaView-v5.5.2/Qt/Components/pqCinemaTrackSelection.cxx
@@ -30,6 +30,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 
 ========================================================================*/
 #include <QDebug>
+#include <QHeaderView>
 
 #include "vtkPVArrayInformation.h"
 #include "vtkPVDataInformation.h"
--- ParaView-v5.5.2.orig/Qt/ApplicationComponents/pqTimeInspectorWidget.cxx
+++ ParaView-v5.5.2/Qt/ApplicationComponents/pqTimeInspectorWidget.cxx
@@ -49,6 +49,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include "vtkSMPropertyHelper.h"
 #include "vtkSMSourceProxy.h"
 
+#include <QHeaderView>
 #include <QLineF>
 #include <QPainter>
 #include <QVariant>
