Index: ParaView-v5.5.2/Qt/ApplicationComponents/pqColorMapEditor.cxx
===================================================================
--- ParaView-v5.5.2.orig/Qt/ApplicationComponents/pqColorMapEditor.cxx
+++ ParaView-v5.5.2/Qt/ApplicationComponents/pqColorMapEditor.cxx
@@ -59,6 +59,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <QDebug>
 #include <QKeyEvent>
 #include <QPointer>
+#include <QStyle>
 #include <QVBoxLayout>
 
 class pqColorMapEditor::pqInternals
Index: ParaView-v5.5.2/Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
===================================================================
--- ParaView-v5.5.2.orig/Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
+++ ParaView-v5.5.2/Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
@@ -43,6 +43,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include "vtkSMUncheckedPropertyHelper.h"
 
 #include <QGridLayout>
+#include <QStyle>
 
 class pqDoubleRangeSliderPropertyWidget::pqInternals
 {
Index: ParaView-v5.5.2/Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
===================================================================
--- ParaView-v5.5.2.orig/Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
+++ ParaView-v5.5.2/Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
@@ -69,6 +69,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include <QPushButton>
 #include <QSet>
 #include <QShortcut>
+#include <QStyle>
 
 //-----------------------------------------------------------------------------
 pqStandardViewFrameActionsImplementation::pqStandardViewFrameActionsImplementation(
Index: ParaView-v5.5.2/Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
===================================================================
--- ParaView-v5.5.2.orig/Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
+++ ParaView-v5.5.2/Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
@@ -39,6 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 #include "vtkSMProxy.h"
 
 #include <QIntValidator>
+#include <QStyle>
 
 class pqViewResolutionPropertyWidget::pqInternals
 {
